libmamba/include/mamba/core/pool.hpp

```diff
@@ -51,9 +38,24 @@ namespace mamba
         void remove_repo(Id repo_id);
 
     private:
-        std::pair<spdlog::logger*, std::string> m_debug_logger;
-        Pool* m_pool;
-        std::list<MRepo> m_repo_list;
+        struct MPoolData;
+
+        Pool* pool();
+        Pool const* pool() const;
+
+        /**
+         * Make MPool behave like a shared_ptr (with move and copy).
```

Comment:
I think the first part of this should be also in a documentation comment at the top of this type.